jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/uper/UperUtil.java

```diff
@@ -1,40 +1,28 @@
 package us.dot.its.jpo.ode.uper;
 
-import java.util.HashMap;
-
+import lombok.extern.slf4j.Slf4j;
 import org.apache.tomcat.util.buf.HexUtils;
 import org.json.JSONObject;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import us.dot.its.jpo.ode.model.OdeMsgPayload;
 import us.dot.its.jpo.ode.util.JsonUtils;
 import us.dot.its.jpo.ode.util.JsonUtils.JsonUtilsException;
 
+import java.util.HashMap;
+
+@Slf4j
 public class UperUtil {
-    private static Logger logger = LoggerFactory.getLogger(UperUtil.class);
-
-    // start flags for BSM, TIM, MAP, SPAT, SRM, SSM, and PSM
-    private static final String BSM_START_FLAG = "0014";
-    private static final String TIM_START_FLAG = "001f";
-    private static final String SPAT_START_FLAG = "0013";
-    private static final String SSM_START_FLAG = "001e";
-    private static final String SRM_START_FLAG = "001d";
-    private static final String MAP_START_FLAG = "0012"; 
-    private static final String PSM_START_FLAG = "0020";
-
-    public enum SupportedMessageTypes {
-        BSM, TIM, SPAT, SSM, SRM, MAP, PSM
+
+    private UperUtil() {
+        throw new UnsupportedOperationException();
     }
 
     // Strips the IEEE 1609.2 security header (if it exists) and returns the payload
-    public static String stripDot2Header(String hexString, String payload_start_flag) {
+    public static String stripDot2Header(String hexString, String payloadStartFlag) throws StartFlagNotFoundException {
         hexString = hexString.toLowerCase();
-        int startIndex = findValidStartFlagLocation(hexString, payload_start_flag);
+        int startIndex = findValidStartFlagLocation(hexString, payloadStartFlag);
         if (startIndex == -1)
-            return "BAD DATA";
-        String strippedPayload = stripTrailingZeros(hexString.substring(startIndex, hexString.length()));
-        return strippedPayload;
+            throw new StartFlagNotFoundException("Start flag" + payloadStartFlag + " not found in message");
+        return stripTrailingZeros(hexString.substring(startIndex));
```

Comment:
note: `hexString.substring(startIndex, hexString.length())` is equivalent to the simpler `hexString.substring(startIndex)` call ([ref](https://www.w3schools.com/java/ref_string_substring.asp))